tests/discovery/test_openalex.py

```diff
@@ -124,3 +124,120 @@ def test_query_error(query_params):
     discoverer = OpenAlex()
     with pytest.raises(QueryError):
         next(discoverer.query(**query_params))
+
+
+def test_focuses_drive_discovery_false():
+    """Test that focuses with drive_discovery=False are skipped."""
+    discoverer = OpenAlex()
+
+    # This should return no results because the focus is skipped
+    results = list(
+        discoverer.query(
+            # Focuses should take precedence over other parameters
+            author="Yoshua Bengio",
+            institution="mila",
+            focuses=Focuses([Focus(type="author", name="Yoshua Bengio", score=1.0)]),
+            page=1,
+            per_page=10,
+        )
+    )
+    assert len(results) == 0
```

Comment:
I'm not sure if the list of results should be empty if all focuses are deactivated for discovery but that is the current behaviour. What do you think?

Reply:
Yeah, it should be empty in that case. The user can just avoid passing the focuses if they want results.

Reply:
Ah ok sorry it's still a bit unclear for me how adding a focus in the config would interact with the discovery feature. Would adding a focus prevent the user from using a "vanilla" discovery feature from the command line? That might be ok I just thought initially that focuses would add better scores to matching results but would still return vanilla results. I'll check the code to understand better